### PR TITLE
Make the uwsgi ini file more configurable with some fallbacks

### PR DESCRIPTION
**Pull request by stephenmoloney**
I want to merge this change because...

I want to make the uwsgi ini file more configurable with some fallbacks for 
more flexible deployments

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.

### CODE REVIEW

**Comment by patrys** [comment]
Is there a key to how the variables/conditionals are sorted? Previously they were all sorted alphabetically, now it seems to be in no particular order.

**Comment by stephenmoloney** [comment]
Yea, I hadn't noticed alphabetical order previously, I will alphabetically reorder.

**Comment by jxltom** [comment]
Using ```UWSGI_PORT``` instead of ```PORT``` will break the compatibility with heroku, since heroku requires web processes must bind to ```$PORT```.

**Comment by stephenmoloney** [comment]
@jxltom 
I'll provide a fallback for $(PORT)

**Comment by stephenmoloney** [comment]
Sorted alphabetically and preserved ${PORT}

Anything more to be done here, we should close this PR otherwise.

**Comment by jxltom** [comment]
Just kindly mention it here that since https://github.com/unbit/uwsgi/pull/1904 is merged into master, after it is released though it is not yet, the logic can be much clearer by using a fallback instead of if-else :D

**Comment by patrys** [comment]
@jxltom Maybe we could put this on hold until then?

**Comment by stephenmoloney** [comment]
@jxltom 
Yea, I spotted that issue, that syntax is preferable. Don't know much about uwsgi release schedule.
Waiting for that release and a refactor for nice format sounds good to me anyways.

**Comment by jxltom** [comment]
@patrys @stephenmoloney This PR is good but not so critical. Without the fallback syntax, the ini file looks somewhat hard to read. So for me I would recommend it can be wait until the new uwsig syntax is released. I hope it can be soon. 🤝

**Comment by maarcingebala** [comment]
@stephenmoloney Would you have time to take a look at this PR again? We'd need to rebase it with `master` and I think it would be ready to merge.

File: Dockerfile [hunk 1]
```diff
@@ -39,7 +39,11 @@ RUN \
 FROM python:3.6-slim
 
 ARG STATIC_URL
-ENV STATIC_URL ${STATIC_URL:-/static/}
+ENV STATIC_URL=${STATIC_URL:-/static/} \
+    PORT=8000 \
+    UWSGI_PROCESSES=4 \
```
Comment (maarcingebala) [inline Dockerfile]: This change renames `PROCESSES` to `UWSGI_PROCESSES`, is that correct?

Comment (maarcingebala) [reply Dockerfile]: Nevermind, I see that it is used properly in `uwsgi.ini`.

**Comment by stephenmoloney** [comment]
sorry @maarcingebala, I started a new role recently and I can't get time at the minute to rebase this. If someone else wants to do that and modify that's fine.

**Comment by maarcingebala** [comment]
Closing due to inactivity.

**Pull request closed without merging**